client/app/pages/queries/query.html

```diff
@@ -74,6 +74,22 @@
             </a>
         </span>
       </h3>
+      <div class="compare-query-version">
+        <span class="zmdi zmdi-archive"></span>
+        <span class="text-muted">Version</span>
+        <strong>{{query.version}}</strong>
+        &gt;
+        <span>Compare to</span>
+        <select id="version-choice">
+          <option ng-repeat="version in versions" value="{{version.object_version}}">
+            {{version.object_version}}
+          </option>
+        </select>
+        <button class="btn btn-s btn-default" ng-click="compareQueryVersion()">
+          <span class="zmdi zmdi-play"></span>
+          Compare
+        </button>
+      </div>
```

Comment:
Regardless of how we implement the preview in the modal, the triggering UI can't be as it is in this pull request. Maybe a middle ground is to have the "updated at" timestamp trigger the comparison UI, and in the modal window show the version picker (instead of having it on the main page). This way the impact is minimal.